comp/logs/agent/config/integration_config.go

```diff
@@ -306,29 +308,45 @@ func (c *LogsConfig) validateTailingMode() error {
 	return nil
 }
 
+// LegacyAutoMultiLineEnabled determines whether the agent has fallen back to legacy auto multi line detection
+// for compatibility reasons.
+func (c *LogsConfig) LegacyAutoMultiLineEnabled(coreConfig pkgconfigmodel.Reader) bool {
+
+	// Handle explicit user initiated fallback to V1
+	if coreConfig.GetBool("logs_config.force_auto_multi_line_detection_v1") {
+		log.Info("Auto multi line detection falling back to legacy mode for log source:", c.Source, "because the force_auto_multi_line_detection_v1 is set to true.")
+		return c.AutoMultiLineEnabled(coreConfig)
+	}
+
+	// Handle transparent fallback if V1 was explicitly configured.
+	if c.AutoMultiLineSampleSize != 0 || coreConfig.GetInt("logs_config.auto_multi_line_default_sample_size") != setup.DefaultLogsLegacySampleSize {
```

Comment:
You can use `IsConfigured` method to know if a setting was configured by the user or is using its default value.

Same for the other 2 if case bellow.

Reply:
TIL about IsConfigured. Thanks, ill try this.